fastly/platform_tls.go

```diff
@@ -0,0 +1,390 @@
+package fastly
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// GetPlatformPrivateKeysInput is an input to the GetPlatformPrivateKeys function.
+// Allowed values for the fields are described at https://docs.fastly.com/api/platform-tls
+type GetPlatformPrivateKeyInput struct {
+	ID string
+}
+
+// PlatformPrivateKeysResponse is a response from the service tls API endpoint
+type PlatformPrivateKeysResponse struct {
+	Status  string                `mapstructure:"status"`
+	Meta    map[string]string     `mapstructure:"meta"`
+	Message string                `mapstructure:"msg"`
+	Data    []*PlatformPrivateKey `mapstructure:"data"`
+}
+
+// PlatformPrivateKeyResponse is a response from the service tls API endpoint
+type PlatformPrivateKeyResponse struct {
+	Status  string              `mapstructure:"status"`
+	Meta    map[string]string   `mapstructure:"meta"`
+	Message string              `mapstructure:"msg"`
+	Data    *PlatformPrivateKey `mapstructure:"data"`
+}
+
+// PlatformPrivateKey .
+type PlatformPrivateKey struct {
+	ID            string                        `mapstructure:"id"`
+	Type          string                        `mapstructure:"type"`
+	Attributes    PlatformPrivateKeysAttributes `mapstructure:"attributes"`
+	Comment       string                        `mapstructure:"comment"`
+	CustomerID    string                        `mapstructure:"customer_id"`
+	CreatedAt     *time.Time                    `mapstructure:"created_at"`
+	UpdatedAt     *time.Time                    `mapstructure:"updated_at"`
+	DeletedAt     *time.Time                    `mapstructure:"deleted_at"`
+	ActiveVersion uint                          `mapstructure:"version"`
+	Versions      []*Version                    `mapstructure:"versions"`
+}
+
+// PlatformPrivateKeysAttributes .
+type PlatformPrivateKeysAttributes struct {
+	KeyLength     string     `mapstructure:"key_length"`
+	KeyType       string     `mapstructure:"key_type"`
+	Name          string     `mapstructure:"name"`
+	CreatedAt     *time.Time `mapstructure:"created_at"`
+	Replace       bool       `mapstructure:"replace"`
+	PublicKeySHA1 string     `mapstructure:"public_key_sha1"`
+}
+
+// GetPlatformPrivateKeys returns stats data based on GetPlatformPrivateKeysInput
+func (c *Client) GetPlatformPrivateKeys() (*PlatformPrivateKeysResponse, error) {
```

Comment:
To follow convention in the rest of the go-fastly package, I'd prefer it if this is called `ListPlatformPrivateKeys()` or similar. The `/tls/private_keys` endpoint is also paginated, so this method should accept a ListPrivateKeysInput struct which models the input parameters `FilterInUse`, `PageNumber` and `PageSize`.